Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -4908,6 +4908,20 @@ RegisterID* BytecodeGenerator::emitYield(RegisterID* argument, JSAsyncGenerator:
     return generatorValueRegister();
 }
 
+RegisterID* BytecodeGenerator::emitAwait(RegisterID* dst, RegisterID* src)
+{
+    ASSERT(dst != ignoredResult());
+    emitYieldPoint(src, JSAsyncGenerator::AsyncGeneratorSuspendReason::Await);
+
+    Ref<Label> normalLabel = newLabel();
+    emitJumpIfTrue(emitEqualityOp<OpStricteq>(newTemporary(), generatorResumeModeRegister(), emitLoad(nullptr, JSGenerator::ResumeMode::NormalMode)), normalLabel.get());
+
+    emitThrow(generatorValueRegister());
+
+    emitLabel(normalLabel.get());
+    return move(dst, generatorValueRegister());
```

Comment:
How about checking `if (dst == ignoredResult())` and skip the move instead of calling `generator.finalDestination` in the caller side?

Reply:
I've removed `generator.finalDestination()` from the caller and instead added handling of `dst == nullptr`, ensuring that no extra `move` is emitted for `ignoredResult()`.

We can just
```cpp
return move(dst, generatorValueRegister());
```

here because `BytecodeGenerator::move()` checks for `ignoredResult`.